### DISCUSSION TITLE
TorchGeo installation on Windows

### DISCUSSION THREAD

**Discussion by roybenhayun** [torchgeo/torchgeo#1538] [Q&A] [answered]
Hi,
first post in torchgeo forum :->

after first experimentations with TorchGeo on Colab notebooks it's time to move to Windows environment for more advanced development. 

I'm using Conda and tried installing either as in https://anaconda.org/conda-forge/torchgeo or also per the instructions in https://torchgeo.readthedocs.io/en/stable/user/installation.html for creating custom conda environment.
But it is hanging for hours on `conda env create --file environment.yml`

would be happy to get a reference on a working windows installation. any advice would be welcomed. 

thanks!

**Comment by adamjstewart** [comment, marked as answer]
In my limited conda experience, conda will either solve quickly or take hours to generate a simple error message explaining why it couldn't solve.

I would personally recommend using pip instead of conda. I actually only use spack, but windows support is still experimental, so I can't recommend that in good faith.

If you figure out what's wrong with the conda environment let us know and we'll try to fix it.

**Reply by roybenhayun** [reply]
thanks Adam. tried with Pip and it works. 
after trying few more things with conda will update again.

thanks!

**Reply by roybenhayun** [reply]
Hi, after trying on few windows machines, conda got stuck every time. when trying the dependencies one by one or batch by batch, it seemed to be hanging specifically when getting to rasterio. would only suggest make the conda limitation a bit clearer in the documentation.

anyway, pip works. and can see that it is used in all tutorials and examples so this is probably the supported option.

thanks!

**Reply by adamjstewart** [reply]
We do have a [note](https://torchgeo.readthedocs.io/en/stable/user/installation.html#id2):

> The installation of torchgeo in this manner is not supported on Windows since pytorch from the conda-forge channel currently does not support Windows. Users are recommended to create a custom conda environment and install torchgeo as shown below.

Could make this a warning so it stands out more.
